proto/frequenz/api/common/v1/lifetime.proto

```diff
@@ -0,0 +1,28 @@
+// Frequenz definition of entity lifetimes, based upon start and end timestamps.
+//
+// Copyright 2023 Frequenz Energy-as-a-Service GmbH
+//
+// Licensed under the MIT License (the "License");
+// you may not use this file except in compliance with the License.
+
+syntax = "proto3";
+
+package frequenz.api.common.v1.lifetime;
```

Comment:
Why wouldn't we put that into microgrid.proto? I don't feel this should deserve its own file.

> Lifetime captures the timestamps indicating the active operational period of a microgrid asset

Reply:
We discussed it to be in `lifetime.proto` since this is being used in several places, including `Component` and `ComponentConnection`.

Reply:
That actually raises a point: should we package microgrid-specific entities in a directory named `microgrid`? That would address the issue with scoping that you brought up.

Otherwise, just putting `Lifetime` in `microgrid.proto`, but leaving the rest out feels arbitrary ..

Reply:
> should we package microgrid-specific entities in a directory named microgrid

we can of course but in this case we should apply the same to all the others. it also raises the question what the structure in these subfolders look like My vote would still got to put it into the microgrid.proto though. At least for now until we have established a clear structure for everything. I don't feel good moving on some parts while not moving on others.

Reply:
As discussed in-person, I updated the file structure for microgrid related entities and grid/delivery_area.

Ready for a look.

I will touch the rest (location, metrics, pagination) in a different PR.

Reply:
This is the structure we have for grid, and microgrid related files:

```
grid/
  delivery_area.proto (package: frequenz.api.common.v1.grid)
microgrid/
  components/
    battery.proto (package: frequenz.api.common.v1.microgrid.components)
    ev_charger.proto (package: frequenz.api.common.v1.microgrid.components)
    fuse.proto (package: frequenz.api.common.v1.microgrid.components)
    grid.proto (package: frequenz.api.common.v1.microgrid.components)
    inverter.proto (package: frequenz.api.common.v1.microgrid.components)
    transformer.proto (package: frequenz.api.common.v1.microgrid.components)
    components.proto (package: frequenz.api.common.v1.microgrid.components)
  sensors/
    sensors.proto (package: frequenz.api.common.v1.microgrid.sensors)
  lifetime.proto (package: frequenz.api.common.v1.microgrid)
```

So, the package of each protobuf file is also its path in the file structure.

Again, `locations.proto`, `metrics.proto`, and `pagination.proto` will be updated in a different PR.

Reply:
I think thats reasonable. I'm still not happy with the lifetime.proto but I don't have a better idea, so lets go with that!